src/shared/backend/services/cloudbuild.ts

```diff
@@ -88,6 +88,37 @@ export const createJob = async (params: JobStatusParams): Promise<Job> => {
   return data as Job;
 };
 
+// Query a job, if it's not found, create it
+export const queryCreateJob = async (params: JobStatusParams): Promise<Job> => {
+  try {
+    return await queryJobStatus(params);
+  } catch (error) {
+    return await createJob(params);
+  }
+};
+
+function timeout(ms: number): Promise<void> {
+  /* eslint-disable-next-line no-promise-executor-return */
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export const queryJobStatusTillSucess = async (
+  params: JobStatusParams
+): Promise<Job | undefined> => {
+  // number of try, 200 * 5s ~= 16.67 min
+  /* eslint-disable no-await-in-loop */
+  for (let i = 0; i < 200; i += 1) {
```

Comment:
Instead of a limited for loop length you could pass in a `timeoutMs` and if we pass that time we break out a while loop?